sdk/core/azure-core/src/main/java/com/azure/core/util/AzureUserAgentUtil.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class for fetching the user agent properties like client library name and version.
+ */
+public class AzureUserAgentUtil {
```

Comment:
We should iterate on the name a little...I think we should make it less specific so that we can load in other properties too.

Reply:
AzureApplicationPropertiesUtil, AzurePropertiesUtil, AzureSdkPropertiesUtil, PropertiesUtil and the other option is to move this method to CoreUtil which already exists.

Reply:
Moved this method to `CoreUtils`.